library/coretests/tests/pin.rs

```diff
@@ -28,7 +28,8 @@ fn pin_const() {
     const fn pin_mut_const() {
         let _ = Pin::new(&mut 2).into_ref();
         let _ = Pin::new(&mut 2).get_mut();
-        let _ = unsafe { Pin::new(&mut 2).get_unchecked_mut() };
+        let r = &mut 2;
+        let _ = unsafe { Pin::new(r).get_unchecked_mut() };
```

Comment:
Out of curiosity, what makes this necessary?

Reply:
You know, thinking about this more, I decided it would be a little less weird to write it as:

```rust
        unsafe {
            let _ = Pin::new(&mut 2).get_unchecked_mut();
        }
```